File: erlang/triangle/1/src/triangle.erl
-module(triangle).

-export([kind/3, test_version/0]).

kind(A, B, C) when A =< 0 ; B =< 0 ; C =< 0 ->
  { error, "all side lengths must be positive" };

kind(A, B, C) when (A + B =< C) ; (A + C =< B) ; (B + C =< A) ->
  { error, "side lengths violate triangle inequality" };

kind(A, B, C) when A =:= B, B  =:= C ->
  equilateral;

kind(A, B, C) when A =:= B ; B =:= C ; C =:= A ->
  isosceles;

kind(A, B, C) when A =/= B ; B =/= C ; C =/= A ->
  scalene.

test_version() -> 1.

File: ruby/triangle/1/triangle.rb

class Triangle
  def initialize(sides)
    @a = sides[0]
    @b = sides[1]
    @c = sides[2]
  end

  def equilateral?
    return false unless triangle?

    @a == @b && @b == @c && @c == @a
  end

  def isosceles?
    return false unless triangle?

    return @a == @b || @b == @c || @c == @a 
  end

  def scalene?
    return false unless triangle?

    @a != @b && @b != @c && @c != @a
  end

  private

  def triangle?
    return false if @a == 0 || @b == 0 || @c == 0

    @a + @b >= @c && @b + @c >= @a && @c + @a >= @b
  end
end
